### PR TITLE
feat(molecule/badgeCounter): add extended variant

### LINKED ISSUE #1450: New Prop for badge counter: Extended
**Is your iteration request related to a problem?**

The [current badge counters](https://sui-components.now.sh/workbench/molecule/badgeCounter/demo) have always a fix width, even when having 2 characters.
It would be awesome to have an extended version

**Describe the solution you'd like**

Either of these two approaches:

A: New prop `width`, with values "fixed" and "responsive"
B: New value "extended" for the current prop `variant`

**Please consider**

The solution has to work with all badges: disabled, selected, etc

**Specs**

[Link to Figma](https://www.figma.com/file/v5yUVkGPtj7YHvIMfJlUdh/SUI-Components-iterations?node-id=118%3A329)

new extended version specs

### PR DESCRIPTION
**Pull request by stivaliserna**
## Molecule/badgeCounter
This closes #1450
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Only
- [ ] Refactor

### Description, Motivation and Context

New badge counter variant called `extended`.

### Screenshots - Animations

Screenshot 2021-04-09 at 12 44 33

### CODE REVIEW

File: components/molecule/badgeCounter/src/index.scss [hunk 1]
```diff
@@ -22,13 +22,15 @@ $molecule-badge-counter-medium-diameter: 16px !default;
 $molecule-badge-counter-medium-diameter-dot: 5px !default;
 $molecule-badge-counter-medium-offset-top: -6px !default;
 $molecule-badge-counter-medium-offset-right: -12px !default;
+$molecule-badge-counter-medium-padding: 0 $p-s !default;
```
Comment (andresin87) [inline components/molecule/badgeCounter/src/index.scss]: $p-molecule-badge-counter-medium

File: components/molecule/badgeCounter/src/index.scss [hunk 2]
```diff
@@ -22,13 +22,15 @@ $molecule-badge-counter-medium-diameter: 16px !default;
...
 
 /* large */
 $molecule-badge-counter-large-diameter: 24px !default;
 $molecule-badge-counter-large-diameter-dot: 6px !default;
 $molecule-badge-counter-large-offset-top: -12px !default;
 $molecule-badge-counter-large-offset-right: -22px !default;
 $fz-molecule-badge-counter-large: $fz-xs !default;
+$molecule-badge-counter-large-padding: 0 $p-base !default;
```
Comment (andresin87) [inline components/molecule/badgeCounter/src/index.scss]: $p-molecule-badge-counter-large

**Comment by andresin87** [comment]
It is changing the width of all sizes for dotted badges

Grabación de pantalla 2021-04-09 a las 15 16 08

**Comment by andresin87** [comment]
why we do not add a small extended demo but medium and large?

File: components/molecule/badgeCounter/README.md [hunk 3]
```diff
@@ -58,6 +58,15 @@ import MoleculeBadgeCounter, {
...
 ```
 
+### Extended Variant
+
+```js
+<MoleculeBadgeCounter
+  variant={moleculeBadgeCounterVariants.EXTENDED}
+  size={moleculeBadgeCounterSizes.MEDIUM}
+/>
+```
```
Comment (andresin87) [inline components/molecule/badgeCounter/README.md:68]: nice!

**Comment by stivaliserna** [comment]
> It is changing the width of all sizes for dotted badges

For the record, we just noticed it's an issue with an update to sui-studio. Not related with changes in this PR.

**Comment by stivaliserna** [comment]
> why we do not add a small extended demo but medium and large?

Because the small badge with a label takes the medium size, and I guess without a label it doesn't make sense to have the extended version.

**Review by andresz1 — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**